sosw/scheduler.py

```diff
@@ -204,6 +221,38 @@ def x_days_back(self, pattern: str) -> List[str]:
         return [str(today - datetime.timedelta(days=num))]
 
 
+    def yesterday(self, pattern: str = 'yesterday') -> List[str]:
+        """
+        Simple wrapper for x_days_back() to return yesterday's date.
+        """
+        assert re.match('yesterday', pattern) is not None, "Invalid pattern {pattern} for `yesterday()`"
+        return self.x_days_back('1_days_back')
+
+
+    def today(self, pattern: str = 'today') -> List[str]:
+        """
+        Returns list with one datetime string (YYYY-MM-DD) equal to today's date.
+        """
+        assert re.match('today', pattern) is not None, "Invalid pattern {pattern} for `today()`"
+        return [str(datetime.date.today())]
+
+
+    def last_week(self, pattern: str = 'last_week') -> List[str]:
+        """
+        Returns list of dates (YYYY-MM-DD) as strings for last week (Sunday - Saturday)
+        :param pattern:
+        :return:
+        """
+        assert re.match('last_week', pattern) is not None, "Invalid pattern {pattern} for `last_week()`"
+
+        today = datetime.date.today()
+        end_date = today - datetime.timedelta(days=today.weekday() + 8)
```

Comment:
This is not really `end_date`. With the current formula it becomes start_date. ;)